ouroboros-consensus/test-storage/Test/Ouroboros/Storage/TestBlock.hs

```diff
@@ -440,10 +481,64 @@ instance UpdateLedger TestBlock where
   ledgerTipPoint = lastAppliedPoint
 
 instance HasAnnTip TestBlock where
-  -- Use defaults
+  type TipInfo TestBlock = IsEBB
+  getTipInfo = thIsEBB . unTestHeader
 
 instance ValidateEnvelope TestBlock where
-  -- Use defaults
+  -- A chain can start with an EBB with block number 0
+  firstBlockNo _ = BlockNo 0
+
+  validateEnvelope cfg oldTip hdr = do
```

Comment:
Store whether we allow EBBs or not in the `BlockConfig` and use that here for a stricter check.